cpp/spherical_volume_rendering_util.cpp

```diff
@@ -0,0 +1,185 @@
+#include "spherical_volume_rendering_util.h"
+#include <vector>
+#include <cmath>
+
+// Takes the absolute value of 'value' and determines if it is less than the tolerance.
+// In this case, we assume tolerance is an arbitrary small value.
+// This is used when floating point mathematics carries rounding error.
+[[nodiscard]] inline bool isNearZero(double value, double tolerance) {
```

Comment:
Do you use any other C++17 features besides `[[nodiscard]]`?

Reply:
Yes unfortunately. It seems that much isn't supported after C++11 (and even features in C++11) for Cython. I'm in the process of Cythonizing the code, and will be removing much of this.